public/JobHandler/index.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { ReactChild } from "react";
+import { EuiLink } from "@elastic/eui";
+import { CoreSetup } from "../../../../src/core/public";
+import { jobSchedulerInstance } from "../context/JobSchedulerContext";
+import { CommonService, IndexService } from "../services";
+import { ReindexJobMetaData, RecoveryJobMetaData } from "../models/interfaces";
+import { ROUTES } from "../utils/constants";
+
+const DetailLink = (props: { index: string }) => {
+  return <EuiLink href={`#${ROUTES.INDEX_DETAIL}/${props.index}`}>{props.index}</EuiLink>;
+};
+
+type TaskResult = {
+  found: boolean;
+  _source: {
+    completed: boolean;
+    response: {
+      failures: {
+        cause?: {
+          reason: string;
+        };
+      }[];
+    };
+    error?: {
+      type: string;
+      reason: string;
+    };
+  };
+};
+
+export const EVENT_MAP = {
+  REINDEX_COMPLETE: "REINDEX_COMPLETE",
+  SPLIT_COMPLETE: "SPLIT_COMPLETE",
+  SHRINK_COMPLETE: "SHRINK_COMPLETE",
+};
+
+const triggerEvent = (eventName: string, data?: unknown) => {
+  const event = new CustomEvent(eventName, {
+    detail: data,
+  });
+  window.dispatchEvent(event);
+};
+
+export const listenEvent = (eventName: string, callback: () => void) => {
+  window.addEventListener(eventName, callback);
+};
+
+export const destroyListener = (eventName: string, callback: () => void) => {
+  window.removeEventListener(eventName, callback);
+};
+
+export function JobHandlerRegister(core: CoreSetup) {
+  const indexService = new IndexService(core.http);
+  const commonService = new CommonService(core.http);
+  jobSchedulerInstance.addCallback({
+    callbackName: "callbackForReindex",
+    callback: async (job: ReindexJobMetaData) => {
```

Comment:
extract this callback into a separate method/function.

Reply:
> 



> Please break down this PR into smaller PRs. It is difficult to review these many files

Sorry, but it is a smaller PR already(We split a huge PR into common/index-management/...) and the common one is hard to split again. We won't make such a big PR next time.

Reply:
break down callbacks to seperate files.
https://github.com/opensearch-project/index-management-dashboards-plugin/commit/e7184e229f17eb720451d569afdf436f245aca04